src/AssemblySharedInfoGenerator/AssemblySharedInfoGenerator.shproj

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>26923a73-d370-43cb-aa1c-f84629ca8d94</ProjectGuid>
+    <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
```

Comment:
so, this gets compiled into each project that references it.... or can you explain it better than MSDN?

Reply:
The source code in this project will get compiled by every project that references it.

Reply:
You can create a shared project in Visual Studio (there is a template for it)
And then you can add source code to it ...as you would for a standard project 
The contents of the project  was generated automatically (I only transfered the .cs and .tt files - which are automatically detected on disk)